### PR TITLE
fix(chat): prevent symlink escape in agent attachments

### PR DESCRIPTION
**Pull request by BunsDev**
### Motivation
- Prevent agent-produced attachment markers from abusing symlinks inside a runtime-granted root to read files outside the grant, which could disclose sensitive data. 
- Restore canonicalization-based containment checks that were present in the previous project-path guard to avoid following symlinks during the allowlist test.

### Description
- Canonicalize the marker path and each runtime-granted root with `fs.realpathSync()` in `resolveAttachmentPath` before performing containment checks, and return the canonical path when inside a grant. 
- Add `isWithinRoot(candidate, root)` helper to centralize the lexical containment test. 
- Preserve existing behavior of silently dropping unresolved or out-of-root markers so `buildAttachment` only reads files that passed the canonicalized containment guard. 
- Add a regression test in `src/lib/server/agent-attachments.test.ts` that creates a symlink inside the allowed root pointing to a file outside the root and asserts the attachment is dropped.

### Testing
- Ran the full API test suite with `pnpm test:api`, and all 115 API test files passed. 
- Ran TypeScript checks with `pnpm exec tsc --noEmit --pretty false`, which completed with no errors. 
- Verified the new unit regression in `src/lib/server/agent-attachments.test.ts` runs under the project test runner as part of `pnpm test:api` and passed.

------
[Codex Task](https://chatgpt.com/codex/cloud/tasks/task_e_6a4a1acf810c8327b9366642efd86b7c)

### CODE REVIEW

File: src/lib/server/agent-attachments.test.ts [hunk 1]
```diff
@@ -97,6 +97,18 @@ try {
...
+    const outsideTextPath = path.join(outside, "secret.env");
+    const symlinkPath = path.join(allowed, "linked-secret.env");
+    await writeFile(outsideTextPath, "TOP_SECRET_TOKEN=symlink_escape");
+    await symlink(outsideTextPath, symlinkPath);
+
+    const text = `nope\n\n\`\`\`coven:attachment\n${JSON.stringify({ path: symlinkPath })}\n\`\`\``;
+    const out = parseAgentAttachments(text, { allowedRoots: [allowed] });
+    assert.equal(out.attachments.length, 0, "symlink target outside allowed root is dropped");
+    assert.equal(out.text, "nope");
+  }
```
[the remark this answers is not in the document]
Comment (CompleteDotTech) [reply src/lib/server/agent-attachments.test.ts:128]: Addressed in c3877d09: the symlink regression now skips only when the platform reports EPERM, EACCES, or ENOSYS, matching the repository’s existing cross-platform symlink-test convention. The case still runs and passes where symlinks are supported.

File: src/lib/server/agent-attachments.ts [hunk 2]
```diff
@@ -84,15 +84,27 @@ function parseMarker(body: string): AttachmentMarker | null {
   };
 }
 
+function isWithinRoot(candidate: string, root: string): boolean {
+  const relative = path.relative(root, candidate);
+  return !relative || (!relative.startsWith("..") && !path.isAbsolute(relative));
+}
```
[the remark this answers is not in the document]
Comment (CompleteDotTech) [reply src/lib/server/agent-attachments.ts:94]: Addressed in c3877d09: the containment check now rejects only `..` and `..${path.sep}` parent traversal, and a regression test confirms an in-root `..notes/notes.txt` path remains allowed.

**Comment by CompleteDotTech** [comment]
Addressed the requested changes in `c3877d09` and resolved the merge conflict with current `main`:

- tightened containment so only actual parent traversal (`..` / `..${path.sep}`) is rejected, while legitimate in-root names such as `..notes/` remain allowed
- preserved the canonicalized symlink-escape guard and the current Turbopack tracing annotations
- made the symlink regression cross-platform by skipping only unsupported/permission-denied symlink creation (`EPERM`, `EACCES`, `ENOSYS`)
- added regression coverage for the `..name` containment edge case

Verification completed:
- `pnpm test:api` — 200 test files passed
- `pnpm exec tsc --noEmit --pretty false` — passed
- full GitHub CI and CodeQL rollup — passed on commit `c3877d09` (including Windows/macOS/Linux cross-environment jobs)

Both review threads were replied to and resolved. The PR is now mergeable with a clean merge state and no outstanding review threads.

**Comment by CompleteDotTech** [comment]
Final revalidation on head `c3877d09` is complete:

- `pnpm test:api` — 200/200 test files passed, including the symlink-escape and `..name` containment regressions
- `pnpm exec tsc --noEmit --pretty false` — passed
- `pnpm check:tests-wired` — all 1050 test files wired (1 documented allowlist)
- `git diff --check origin/main...HEAD` — clean
- current-main merge simulation — no conflicts
- full GitHub CI and CodeQL rollup — all checks passed across Linux, Windows, and macOS
- review threads — both actionable threads replied to and resolved; no outstanding requests

The acceptance and testing criteria are satisfied. No additional code change or empty commit was needed after the existing fix commit; the PR remains `MERGEABLE` / `CLEAN`.
